src/components/__tests__/results.js

```diff
@@ -0,0 +1,110 @@
+import Vue from 'vue';
+import { Results } from 'vue-instantsearch';
+
+test('renders proper HTML', () => {
+  const searchStore = {
+    results: [
+      { objectID: 1 },
+      { objectID: 2 },
+      { objectID: 3 },
+      { objectID: 4 },
+    ],
+  };
+  const Component = Vue.extend(Results);
+  const vm = new Component({
+    propsData: {
+      searchStore,
+    },
+  });
+  vm.$mount();
+
+  expect(vm.$el.outerHTML).toMatchSnapshot();
+});
+
+test('should be hidden if not results are yielded', () => {
+  const searchStore = {
+    results: [],
+  };
+  const Component = Vue.extend(Results);
+  const vm = new Component({
+    propsData: {
+      searchStore,
+    },
+  });
+  vm.$mount();
+
+  expect(vm.$el.outerHTML).toMatchSnapshot();
+});
+
+test('should allow to override result rendering', () => {
+  const searchStore = {
+    results: [{ objectID: 1 }, { objectID: 2 }],
+  };
+
+  const Component = Vue.extend({
+    render(h) {
+      return h('ais-results', {
```

Comment:
I don't think it will make this more readable.
Also I think that would require `babel-jsx`.
And currently Vue's JSX format is kinda different from the Vue one: https://github.com/vuejs/babel-plugin-transform-vue-jsx#difference-from-react-jsx

I'll probably try it out a bit when switching to babel.